tests/x11regressions/gnomecase/application_starts_on_login.pm

```diff
@@ -0,0 +1,117 @@
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+use base "x11test";
+use strict;
+use testapi;
+
+#testcase 5255-1503973: Gnome: Applications starts on login 
+
+sub tweak_startupapp_menu {
+    my ($string) = @_;
+
+    wait_screen_change {
+        send_key "super";
+    };
+    type_string "settings", 1;
+    send_key "ret";
+    assert_screen "gnome-settings";
+    type_string "tweak", 1;
+    send_key "ret";
+    assert_screen "tweak-tool";
+    send_key_until_needlematch "tweak-startapp", "down";
+}
+
+sub logout_and_login {
+    my ($string) = @_;
+
+    assert_and_click "system-indicator";
+    assert_and_click "user-logout-sector";
+    assert_and_click "logout";
+    send_key "ret";
+    assert_screen "displaymanager";
+    send_key "ret";
+    type_string "$password";
+    send_key "ret";
+}
+
+sub alter_status_auto_save_session {
+    my ($string) = @_;
+
+    send_key "super";
+    type_string "settings", 1;
+    send_key "ret";
+    assert_screen "gnome-settings";
+    type_string "dconf", 1;
+    send_key "ret";
+    send_key_until_needlematch "dconf-org", "down";
+    assert_and_click "unfold";
+    send_key_until_needlematch "dconf-org-gnome", "down";
+    assert_and_click "unfold";
+    send_key_until_needlematch "dconf-gnome-evolution", "down";
+    assert_and_click "scroll-down"; #this step aim to work around screen not scroll down automate issue
+    send_key_until_needlematch "gnome-session", "down";
+    assert_and_click "auto-save-session";
+    send_key "alt-f4";
+    send_key "alt-f4";
+}
+
+sub run() {
+    my $self = shift;
+    
+    #add firefox to startup application
+    tweak_startupapp_menu;
+    assert_and_click "tweak-startapp-add";
+    assert_screen "tweak-startapp-applist";
+    send_key_until_needlematch "applicationstart-firefox", "down";
+    assert_and_click "tweak-addapp-button";
+    assert_screen "startapp-firefox-added";
+    wait_screen_change {
+        send_key "alt-f4";
+    };
+    send_key "alt-f4";
+
+    logout_and_login;
+    assert_screen "application-running-firefox";
+    send_key "alt-f4";
+
+    #remove firefox from startup application
+    tweak_startupapp_menu;
+    assert_screen "startapp-firefox-added";
+    assert_and_click "startapp-delete";
+    send_key "alt-f4";
+    send_key "alt-f4";
+    assert_screen "generic-desktop", 5;
+
+    logout_and_login;
+    assert_screen "generic-desktop", 5;
+
+    #set auto-save-session;
+    ##For got information: start from gnome 3, 
+    ##for lacking maintainence, 
+    ##auto-save-session functionality has been abandoned; 
+    ##current status: just firefox works
+    ##so in the future will consider remove openqa code for this session
+    alter_status_auto_save_session;
+
+    assert_and_click "application-menu";
+    wait_screen_change {
+        assert_and_click "application-menu-firefox";
+    };
+    sleep 60;
```

Comment:
can you replace the `sleep 60` by a useful `assert_screen` or `wait_serial`? I am sure the `sleep 60` means you are waiting for some event.